processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultColumnSelectorFactoryMaker.java

```diff
@@ -156,23 +160,26 @@ public ColumnValueSelector makeColumnValueSelector(@Nonnull String columnName)
     @Override
     public ColumnCapabilities getColumnCapabilities(String column)
     {
-      return withColumnAccessor(column, columnAccessor ->
-          new ColumnCapabilitiesImpl()
+      return withColumnAccessor(column, columnAccessor -> {
+        if (columnAccessor == null) {
+          return null;
+        } else {
+          return new ColumnCapabilitiesImpl()
               .setType(columnAccessor.getType())
               .setHasMultipleValues(false)
               .setDictionaryEncoded(false)
-              .setHasBitmapIndexes(false));
+              .setHasBitmapIndexes(false);
+        }
+      });
     }
 
     private <T> T withColumnAccessor(String column, Function<ColumnAccessor, T> fn)
     {
+      @Nullable
       ColumnAccessor retVal = accessorCache.get(column);
       if (retVal == null) {
         Column racColumn = rac.findColumn(column);
-        if (racColumn == null) {
-          throw DruidException.defensive("Column[%s] not found!", column);
-        }
-        retVal = racColumn.toAccessor();
+        retVal = racColumn == null ? null : racColumn.toAccessor();
```

Comment:
Reading this code has made me realize that we need to do a `containsKey()` check in order to actually take advantage of the cache.  I.e. if we cache that it was `null`, we are going to keep asking and overwriting the cache for that column because we are checking the result of `.get()` instead of `.containsKey()`.

I'm leaving this comment so that we have a shared understanding, not because I'm asking it to be adjusted in this PR.